ios/testmanagerd/xcuitestrunner.go

```diff
@@ -358,18 +359,26 @@ func runXUITestWithBundleIdsXcode15Ctx(
 	}
 	defer appserviceConn.Close()
 
-	pid, err := startTestRunner17(device, appserviceConn, "", testRunnerBundleID, strings.ToUpper(testSessionID.String()), info.testApp.path+"/PlugIns/"+xctestConfigFileName, args, env)
+	testRunnerLaunch, err := startTestRunner17(device, appserviceConn, "", testRunnerBundleID, strings.ToUpper(testSessionID.String()), info.testApp.path+"/PlugIns/"+xctestConfigFileName, args, env)
 	if err != nil {
 		return make([]TestSuite, 0), fmt.Errorf("runXUITestWithBundleIdsXcode15Ctx: cannot start test runner: %w", err)
 	}
 
+	defer testRunnerLaunch.Close()
+	go func() {
+		_, err := io.Copy(testListener.logWriter, testRunnerLaunch)
```

Comment:
Does this imply an infinite loop? How is it going to behave when `Read()` on `testRunnerLaunch` fails with an error different than `io.EOF`?

Reply:
It copies all there is available until an error happens, and it returns the error if it's not `io.EOF` ([docs](https://pkg.go.dev/io#Copy))

for us that means we get a log if a real error happens. I don't think there's much more we can do here as a new connection to the stdio service will have a new UDID and we can't change that on the existing app. And killing the whole test run because of this sound also a bit harsh (it could be a config option for the test run, but I don't want to add another argument to the methods here for that)